lib/domain.js

```diff
@@ -73,12 +74,14 @@ const asyncHook = createHook({
     if (process.domain !== null && process.domain !== undefined) {
       // If this operation is created while in a domain, let's mark it
       pairing.set(asyncId, process.domain[kWeak]);
-      ObjectDefineProperty(resource, 'domain', {
-        configurable: true,
-        enumerable: false,
-        value: process.domain,
-        writable: true
-      });
+      if (type !== 'PROMISE' || resource instanceof Promise) {
```

Comment:
You should add a comment on what this logic does. It is not immediately obvious.